core/src/main/java/org/apache/calcite/config/CalciteSystemProperty.java

```diff
@@ -429,7 +430,8 @@ private static Properties loadProperties() {
         Thread.currentThread().getContextClassLoader(),
         CalciteSystemProperty.class.getClassLoader());
     // Read properties from the file "saffron.properties", if it exists in classpath
-    try (InputStream stream = classLoader.getResourceAsStream("saffron.properties")) {
```

Comment:
Why do we need to add these restrictions？ I noticed a lot of changes in this.

Reply:
Thanks for the review @NobiGo, these changes is because we upgraded guava version, which checks nullability more strictly, many apis added nullability annotation.